circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/DropTableService.java

```diff
@@ -25,28 +26,62 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.hotels.bdp.circustrain.api.data.DataManipulator;
 import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+import com.hotels.hcommon.hive.metastore.iterator.PartitionIterator;
 
 public class DropTableService {
 
-  private static final Logger LOG = LoggerFactory.getLogger(DropTableService.class);
+  private static final Logger log = LoggerFactory.getLogger(DropTableService.class);
   private static final String EXTERNAL_KEY = "EXTERNAL";
   private static final String IS_EXTERNAL = "TRUE";
+  private static final short BATCH_SIZE = (short) 1000;
 
   /**
    * Removes all parameters from a table before dropping the table.
+   * 
+   * @throws Exception if the table can't be deleted.
    */
-  public void removeTableParamsAndDrop(
+  public void dropTable(CloseableMetaStoreClient client, String databaseName, String tableName) throws Exception {
```

Comment:
I think this and the changes below are much clearer now, also better using the partition iterator properly, nice.